File: Proyecto 1/Proyecto1OLC/src/AST/Nodo.java
package AST;

import java.util.ArrayList;

/**
 *
 * @author TheJhonX
 */
public class Nodo {
    private String nombre;
    private ArrayList<Nodo> hijos;
    private String valor;
    private int idNodo; 

    public Nodo(String nombre) {
        this.nombre = nombre;
        hijos = new ArrayList<>();
        this.idNodo = 0;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public ArrayList<Nodo> getHijos() {
        return hijos;
    }

    public void setHijos(ArrayList<Nodo> hijos) {
        this.hijos = hijos;
    }

    public String getValor() {
        return valor;
    }

    public void setValor(String valor) {
        this.valor = valor;
    }

    public int getIdNodo() {
        return idNodo;
    }

    public void setIdNodo(int idNodo) {
        this.idNodo = idNodo;
    }
    
    public void addHijo(Nodo hijo){
        hijos.add(hijo);
    }
    
    
}

File: Proyecto 2/server/src/AST/Nodo.ts
export class Nodo{

    private hijos:any[];
    private valor:string;
    private idNodo:number;

    constructor(public nombre:string){
        this.hijos = [];
        this.valor = "";
        this.idNodo = 0;
    }

    public getNombre():string{
        return this.nombre;
    }

    public setNombre(nombre:string){
        this.nombre = nombre;
    }

    public getHijos():any[]{
        return this.hijos;
    }

    public setHijos(hijos:any[]){
        this.hijos = hijos;
    }

    public getValor():string{
        return this.valor;
    }

    public setValor(valor:string){
        this.valor = valor;
    }

    public getIdNodo():number{
        return this.idNodo;
    }

    public setIdNodo(id:number){
        this.idNodo = id;
    }

    public addHijo(hijo:Nodo){
        this.hijos.push(hijo);
    }
}